test/utils/wasm3_engine.cpp

```diff
@@ -81,6 +81,9 @@ void Wasm3Engine::set_memory(fizzy::bytes_view memory)
 
     uint32_t size;
     const auto data = m3_GetMemory(m_runtime, &size, 0);
+    if (data == nullptr)
+        return;
```

Comment:
I consider these a failure. Non-empty memory was supplied but it can't be overriden.

Reply:
I don't mind agreeing on this though, @chfast ?

Reply:
Properly failing would be better than crashing, so we could make it return `false` in this case.

Reply:
I think it should work like this:
1. If the instance has memory, it should be reset accordingly to the `memory` argument. This should be done also for `memory` being empty (current checks are wrong here).
2. If the instance has no memory AND the `memory` argument is not empty -> failure.

Reply:
Hm looks like wasm3's API doesn't allow to reset the memory to empty... Best i can do is overwrite it with zeroes.